controllers/sriov-controller/dpapi-controller.go

```diff
@@ -244,24 +232,89 @@ func (s *serviceInstanceController) ListAndWatch(e *pluginapi.Empty, d pluginapi
 	}
 }
 
+// vfioConfig is stuct used to store vfio device specific information
+type vfioConfig struct {
+	VFIODevice string `yaml:"vfioDevice" json:"vfioDevice"`
+	PCIAddr    string `yaml:"pciAddr" json:"pciAddr"`
+}
+
+// getVFIODevSpecs looks for vfio device's specs, example PCI address and return vfioConfig instance.
+// In future other parameters could be added, example NUMATopology, CPUPinning, etc.
+func (s *serviceInstanceController) getVFIODevSpecs(vfioDev string) (vfioConfig, error) {
+	for _, vfio := range s.serviceInstance.vfs {
+		if vfio.VFIODevice == vfioDev {
+			return vfioConfig{
+				VFIODevice: vfioDev,
+				PCIAddr:    vfio.PCIAddr,
+			}, nil
+		}
+	}
+	return vfioConfig{}, fmt.Errorf("vfio %s device is not found", vfioDev)
+}
+
 // Allocate which return list of devices.
 func (s *serviceInstanceController) Allocate(ctx context.Context, reqs *pluginapi.AllocateRequest) (*pluginapi.AllocateResponse, error) {
 	logrus.Infof("network service %s received Allocate from kubelet", s.networkServiceName)
+	// Allocating Device Plugin API response struct
 	responses := pluginapi.AllocateResponse{}
+	// Allocating slice of vfioConfigs, the content of this slice will be saved as a json file and make available to POD requesting
+	// the network service.
+	vfioDevs := []vfioConfig{}
+	// Bulding per network service key for Env variable, it will point to the network service configuration
+	// file.
+	networkServiceName := strings.ToLower(strings.Split(s.networkServiceName, "/")[1])
+	networkServiceName = strings.Replace(networkServiceName, "-", "_", -1)
+	if strings.HasPrefix(networkServiceName, "sriov_") {
+		networkServiceName = strings.Split(networkServiceName, "sriov_")[1]
+	}
+	key := "NSM_VFS_" + networkServiceName
```

Comment:
Should `NSM_VFS_` be a constant at the top of the file?

Reply:
Done